wandb/apis/importers/wandb.py

```diff
@@ -0,0 +1,1133 @@
+            d[_project] = list(placeholder_artifacts(_project))
+
+        with progress.live:
+            outer_task = progress.task_pbar.add_task(
+                "Tidy project artifacts", total=len(d)
+            )
+            for name, arts in d.items():
+                task = progress.subtask_pbar.add_task(
+                    f"Remove placeholders {name}", total=len(arts)
+                )
+                for art in arts:
+                    try:
+                        art.delete(delete_aliases=True)
+                    except Exception as e:
+                        if "cannot delete system managed artifact" not in str(e):
+                            raise e
+                    progress.subtask_pbar.update(task, advance=1)
+                progress.subtask_pbar.update(task, visible=False)
+                progress.task_pbar.update(outer_task, advance=1)
+
+    def use_artifact_sequence(
+        self, sequence: ArtifactSequence, config: Optional[ImportConfig] = None
+    ) -> None:
+        """Do the equivalent of `run.use_artifact(art)` for each artifact in the artifact sequence.
+
+        Use `config` to specify alternate settings like where the artifact sequence should be used
+        """
+        if config is None:
+            config = ImportConfig()
+
+        settings_override = {
+            "api_key": self.dst_api_key,
+            "base_url": self.dst_base_url,
+            "resume": "true",
+            "resumed": True,
+        }
+
+        send_manager_config = internal.SendManagerConfig(
+            use_artifacts=True,
+            debug_mode=config.debug_mode,
+        )
+
+        sequence = list(sequence)
+        s = sequence[0]
+        _type = s.type
+        name, _ = s.name.split(":")
+
+        task = progress.subtask_pbar.add_task(
+            f"Use Artifact Sequence ({_type}/{name})", total=len(sequence)
+        )
+        for art in sequence:
+            if art.type == "job":
+                # Job is a special type that can't be used yet
+                continue
+
+            wandb_runs = art.used_by()
+            if wandb_runs == []:
+                # Don't try to download an artifact that doesn't exist
+                continue
+
+            try:
+                path = art.download()
+            except Exception as e:
+                wandb_logger.error(
+                    f"Error downloading artifact {art} -- {e}",
+                    extra={
+                        "entity": wandb_runs[0].entity,
+                        "project": wandb_runs[0].project,
+                        "run_id": wandb_runs[0].id,
+                    },
+                )
+                continue
+
+            new_art = make_new_art(art)
+
+            if Path(path).is_dir():
+                new_art.add_dir(path)
+
+            for wandb_run in wandb_runs:
+                run = WandbRun(wandb_run)
+                internal.send_artifacts_with_send_manager(
```

Comment:
Nit: I guess this works, but you don't have to create the artifact again just to mark it as used by the run. You can do `Run.use_artifact()` ([docs](https://docs.wandb.ai/ref/python/public-api/run#use_artifact)).

Reply:
I don't think this re-uploads the artifact if it already exists